src/panels/lovelace/cards/hui-card.ts

```diff
@@ -0,0 +1,141 @@
+import { PropertyValues, ReactiveElement } from "lit";
+import { customElement, property, state } from "lit/decorators";
+import { MediaQueriesListener } from "../../../common/dom/media_query";
+import "../../../components/ha-svg-icon";
+import { LovelaceCardConfig } from "../../../data/lovelace/config/card";
+import type { HomeAssistant } from "../../../types";
+import { computeCardSize } from "../common/compute-card-size";
+import {
+  attachConditionMediaQueriesListeners,
+  checkConditionsMet,
+} from "../common/validate-condition";
+import { createCardElement } from "../create-element/create-card-element";
+import type { Lovelace, LovelaceCard, LovelaceLayoutOptions } from "../types";
+
+@customElement("hui-card")
+export class HuiCard extends ReactiveElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @property({ attribute: false }) public lovelace!: Lovelace;
+
+  @state() public _config?: LovelaceCardConfig;
+
+  private _element?: LovelaceCard;
+
+  private _listeners: MediaQueriesListener[] = [];
+
+  protected createRenderRoot() {
+    return this;
+  }
+
+  public disconnectedCallback() {
+    super.disconnectedCallback();
+    this._clearMediaQueries();
+  }
+
+  public connectedCallback() {
+    super.connectedCallback();
+    this._listenMediaQueries();
+    this._updateElement();
+  }
+
+  public getCardSize(): number | Promise<number> {
+    if (this._element) {
+      const size = computeCardSize(this._element);
+      return size;
+    }
+    return 1;
+  }
+
+  public getLayoutOptions(): LovelaceLayoutOptions {
+    const configOptions = this._config?.layout_options ?? {};
+    if (this._element) {
+      const cardOptions = this._element.getLayoutOptions?.() ?? {};
+      return {
+        ...cardOptions,
+        ...configOptions,
+      };
+    }
+    return configOptions;
+  }
+
+  public setConfig(config: LovelaceCardConfig): void {
+    if (this._config === config) {
+      return;
+    }
+    this._config = config;
+    this._element = createCardElement(config);
```

Comment:
`hui-card` should probably also handle the rebuild logic, we can then remove it anywhere else.